app/src/main/java/org/mozilla/fenix/utils/StartupTaskManager.kt

```diff
@@ -0,0 +1,40 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.utils
+
+typealias StartupTask = () -> Unit
+
+/**
+ * A queue of tasks that are performed at specific points during Fenix startup.
+ *
+ * This queue contains a list of startup tasks. Each task in the queue will be started once Fenix
+ * is visually complete (as long as it has not be cancelled since it was added to the queue).
+ * Each task will execute on the main thread.
```

Comment:
nit: this comment should probably go where `add` is because:
1) devs are more likely to see method kdoc than the class kdoc
2) if we change the behavior, it's very likely `add` will let devs specify what thread it should run on